### PR TITLE
Igor Yermak - W.I.P

### PR DESCRIPTION
**Pull request by didacta**
- [x] added dependencies and server listen

- [x] ANSWERS.md complete
 
- [x] GET call to endpoint

- [x] Compare previous and today's BTC value

### CODE REVIEW

File: ANSWERS.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+##Answers 
+
+1. Node is a backend framework that was developed to allow developers to run Javascript on the server side of an application. This is very useful because previously J.S was limited to the front end and now developers have an easier time transitioning to full-stack.
+
+2. Express is a framework that helps manage routes in an application that is developed for the server side. Express is what lets you handle API requests and middleware.  
+
+3. Middleware is much what it sounds like, code that allows you to connect different parts of your application to allow them all to work together and know what the other parts are doing.
```
Comment (dys2) [inline ANSWERS.md]: I like to think of it as a function that receives data, does something to that data, and passes it along

File: src/server.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          if (yesterday > today) {
+            const difference = yesterday - today;
+            const message = 'Compared to yesterday, Bitcoin is down $' + difference + ':(';
+            res.status(STATUS_OK);
+            res.send(message);
+          } else if (yesterday < today) {
+            difference = today - yesterday;
+            const message = "Compared to yesterday, Bitcoin is up $" + difference + ':)';
+            res.status(STATUS_OK);
+            res.send(message);
```
Comment (dys2) [inline src/server.js]: I like how you are personalizing the message based on whether you the price is up or down but some unnecessary code is being repeated

File: src/server.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+server.get('/compare', (req, res) => {
+
+
+  fetch(YESTERDAY_PRICE_REQ)
+    .then(data => data.json())
+  console.log(data.json())
+    .then(data => {
+      const yesterdayOriginal = data.bpi[Object.keys(data.bpi)[0]];
+      const yesterday = yesterdayOriginal.toFixed(3);
+      fetch(CURRENT_PRICE_REQ)
```
Comment (dys2) [inline src/server.js]: can avoid nesting by using Promise.All

File: src/server.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
+const express = require('express');
+//const bodyParser = require('body-parser');
```
Comment (dys2) [inline src/server.js]: no need for body parser since we're already dealing with json

File: src/server.js [hunk 5]
```diff
@@ -1,53 +1,80 @@
...
           if (yesterday > today) {
             const difference = yesterday - today;
-            const message = 'Compared to yesterday, Bitcoin is down $' + difference + ':(';
+            const message = `Compared to yesterday, Bitcoin is down $${difference} :(`;
             res.status(STATUS_OK);
             res.send(message);
           } else if (yesterday < today) {
             difference = today - yesterday;
-            const message = "Compared to yesterday, Bitcoin is up $" + difference + ':)';
+            const message = `Compared to yesterday, Bitcoin is up $${difference} :)`;
```
Comment (johnoro) [inline src/server.js]: To expand on the last comment in regards to this code, you could break out the stuff that's the same outside of the `if...else if`; for example:
```javascript
let difference = today - yesterday;
let message = 'Compared to yesterday, Bitcoin is ';
if (yesterday > today) {
  difference = yesterday - today;
  message += `down $${difference} :(`;
} else if (yesterday < today) {
  message += `up $${difference} :)`;
}
res.status(STATUS_OK);
res.send(message);
```
You also might want to include an `else` for the potential case that today's value is the same as yesterday's.

File: src/server.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+ 
+.next(()) => {
+    res.status(STATUS_OK);
+    res.json({difference : currentPrice - yesterdayPrice});
+    
+  })
+  .catch((err) => {
+  res.status(STATUS_ERROR);
+  res.send('Error fetching prices');
+  });
```
Comment (johnoro) [inline src/server.js]: It's best to get rid of old code over commenting it out so as to not make a mess out of your files.

**Pull request closed without merging**